### PR TITLE
case 17773: reduce likelihood MyAvatar will get stuck in mesh geometry when flying around

### PR DESCRIPTION
**Pull request by AndrewMeadows**
https://highfidelity.manuscript.com/f/cases/17773/Fast-flying-will-consistently-allow-the-user-to-clip-into-multiple-walls-and-they-are-stuck-until-domain-reloads

This PR doesn't solve or prevent all snags between MyAvatar and mesh geometry but it tries to make it harder without adding too many bad side effects.

### CODE REVIEW

**Comment by AndrewMeadows** [comment]
**To test:**
**(1)** Login to TheSpot
**(2)** Fly around with the SHIFT key (faster!) and collide with walls of the models.  The goal it to try to get "stuck" in the middle of the walls.
**(3)** While it may be possible to get stuck using this PR it should happen less often and it should be easier to extricate yourself... when flying.  Because `MyAvatar`'s RigidBody will not collide against backfacing triangles when it is flying.   Collisions with backfacing triangles **will still work** when the avatar is NOT flying.
Note: it may be easier for a flying `MyAvatar` to tunnel completely through mesh geometry and out the other side.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16871/HighFidelity-Beta-PR14795-e372cb6.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16871/HighFidelity-Beta-PR14795-e372cb6.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16871/HighFidelity-Beta-Interface-PR14795-e372cb6.exe

**Comment by AndrewMeadows** [comment]
Note, this work is a combination of #13711 (which was removed after it introduced a bug when avatar walks along the ground, see rollback #13840) and #13864 (which was never merged because it didn't fully fix the problem).  The problem in 13711 is avoided here by only enabling the feature when the avatar is flying, and the 13864 work was good... but it just wasn't a 100% fix.

File: libraries/physics/src/CharacterController.cpp [hunk 1]
```diff
@@ -124,6 +124,11 @@ void CharacterController::setDynamicsWorld(btDynamicsWorld* world) {
             _rigidBody->setGravity(_currentGravity * _currentUp);
             // set flag to enable custom contactAddedCallback
             _rigidBody->setCollisionFlags(btCollisionObject::CF_CUSTOM_MATERIAL_CALLBACK);
+
+            // enable CCD
+            _rigidBody->setCcdSweptSphereRadius(_radius);
+            _rigidBody->setCcdMotionThreshold(_radius);
```
Comment (AndrewMeadows) [inline libraries/physics/src/CharacterController.cpp]: Enabling CCD makes it harder (but not impossible) for the Body to tunnel through thin mesh.

File: interface/src/Application.cpp [hunk 2]
```diff
@@ -6289,6 +6289,8 @@ void Application::update(float deltaTime) {
                 avatarManager->handleProcessedPhysicsTransaction(transaction);
 
                 myAvatar->prepareForPhysicsSimulation();
+                _physicsEngine->enableGlobalContactAddedCallback(myAvatar->isFlying());
+
```
Comment (AndrewMeadows) [inline interface/src/Application.cpp]: We enable/disable the custom contact filter when `MyAvatar` is flying/walking respectively.

**Comment by JeffClinton** [comment]
Build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17093/HighFidelity-Beta-PR14795-e372cb6.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17093/HighFidelity-Beta-PR14795-e372cb6.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17093/HighFidelity-Beta-Interface-PR14795-e372cb6.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17093/HighFidelity-Beta-PR14795-e372cb6.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17093/HighFidelity-Beta-Interface-PR14795-e372cb6.exe

**Comment by JeffClinton** [comment]
Currently testing

**Pull request merged**